### ISSUE TITLE
finance.iban does not create valid iban

### ISSUE THREAD

**Issue by napres1001** [faker-js/faker#1832]
### Pre-Checks

- [X] Follow our [Code of Conduct](https://github.com/faker-js/faker/blob/next/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/faker-js/faker/blob/next/CONTRIBUTING.md).
- [X] Read the [docs](https://fakerjs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/faker-js/faker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Faker issue and not related to a combination with another package.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/faker-js/faker/discussions) or join our [Discord Chat Server](https://chat.fakerjs.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.
- [ ] I am willing to provide a PR.

### Describe the bug

when using:
"finance.iban" the iban is always invalid.
In my case i used:
faker.finance.iban(false, 'DE')
and got for example:
DE88 1533 9441 3800 0600 27

but even the iban mentioned in the documentation are not valid:
https://fakerjs.dev/api/finance.html#iban

to check, if a iban is valid you can use:
https://www.ibancalculator.com/iban_validieren.html

### Minimal reproduction code

https://stackblitz.com/edit/faker-js-demo-ngipzd?embed=1&file=index.html

### Environment Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 12th Gen Intel(R) Core(TM) i5-1245U
    Memory: 3.17 GB / 15.44 GB
  Binaries:
    Node: 18.10.0 - C:\Program Files\nodejs\node.EXE
    npm: 2.15.12 - ~\source\repos\wl-cypress-smoke-test\node_modules\.bin\npm.CMD
    npm version 9.4.2
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (109.0.1518.70)
    Internet Explorer: 11.0.19041.1566
    Google Chrome: 109.0.5414.120
    Mozialle Firefox: 109.0.1

faker 7.6.0
by npm install @faker-js/faker --save-dev
```

### Which module system do you use?

- [ ] CJS
- [ ] ESM

### Used Package Manager

npm

**Comment by ST-DDT** [comment]
From the page you linked:

![grafik](https://user-images.githubusercontent.com/1579362/218545386-5e423af4-8abe-4a88-a5e2-3f30f89540f1.png)

Do you have a list of allowed bank codes?

**Comment by napres1001** [comment]
i cant provide a full list of valid bank-codes. maybe there is some online source (API?) for it?
like:
https://www.iban.com/validation-api

**Comment by b-pagis** [comment]
**TL;DR;**

This is a very hard task to accomplish and it would require constant maintenance to keep the correctness. IMHO, implementing country specific checks are, probably, not worth doing as it would only be required for very specific use cases.

**Here is a longer explanation, why it is not worth doing**

IBAN has generic format (I'll use the one from this issue):
- DE - country code
- 88 - checksum
- 153394413800060027 - BBAN

Each BBAN has a format, which is different for each country. Here is a [page](https://www.mobilefish.com/services/random_iban_generator/random_iban_generator.php) that very well illustrates what format each country is using.

In this case BBAN is divided to following codes:
- 15339441 - branch code (or bank code)
- 3800060027 - account number

Taken from service description, it checks account checksum (explanation taken from service);
> Methode 06, Konto 0335652860, BLZ 87096214, Prüfziffer 0 steht an Position 10, erwartete Prüfziffer: 0. Überblick über die Berechnung: Nimm die Ziffern auf den Positionen 1 bis 9 - hier: 033565286 -, multipliziere sie von rechts nach links mit den Gewichten 2,3,4,5,6,7,2,3,4, addiere die Produkte, bilde den Rest der Division durch 11, ziehe das Ergebnis von 11 ab, und das Ergebnis modulo 11 ist die erwartete Prüfziffer.

Which, translates to something like this (used google translation):
> Method 06, account 0335652860, bank code 87096214, check digit 0 is in position 10, expected check digit: 0. Overview of the calculation: Take the digits in positions 1 to 9 - here: 033565286 -, multiply them from right to left by the weights 2,3,4,5,6,7,2,3,4, add the products, form the remainder of division by 11, subtract the result from 11, and the result modulo 11 is the expected check digit.

Some countries have their own specific additional validation logic to check if account is correct.  Here is [page](https://docs.oracle.com/cd/E18727_01/doc.121/e13483/T359831T498954.htm) form oracle docs explaining it.

Furthermore, to pass online service checks, it would require to use real life data, which is constantly changing. This is what ibancalculator service is doing. It actually checks if the sort code 15339441 (first 8 symbols of bban) exists in the National Competent Authorities in charge of banking lists. In this case it is bundesbank.de. 

DE NCA allows to download the list of code - [link to the page](https://www.bundesbank.de/en/tasks/payment-systems/services/bank-sort-codes/download-bank-sort-codes-626218) or search it via [web interface](https://www-bundesbank-de.translate.goog/action/de/747634/bbkpspsearch?freeSearch&hitsPerPage=10&city&bankRoutingNumber&panNr&sort=bankRoutingNumber+asc,+maintainsBRN+desc,+city+asc,+maintainsBRN+desc&type=simple&pageBlock=0&title&bic&_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=wapp).

My knowledge in this field is a little bit outdated, but couple years ago there were some NCAs that would only allowed to get the lists via web interface without the possibility to download them in any other format. Even heard some talks that people had to scrape some NCAs pages in order to get the list and keep it updated. Given the nature of NCAs (part of the government), I do not expect that changes are quickly applied in their systems. But, I could be wrong as there were few grand changes in EU banking sector that could have improved how the things works.

So, honestly, if you have a specific case, where you would need to check on real life data, it would probably make more sense to take the record from NCA list and construct the IBAN account around it, rather than implementing all this logic into the library. But again, this is only my opinion ;)

**Comment by Qaklaker** [comment]
There is a Website, which generates valid IBANs. Maybe you can work together with them to integrate that logic into Faker. That would be awesome

Mentioned Website is:
http://www.randomiban.com/?country=Italy

**Comment by matthewmayer** [comment]
> There is a Website, which generates valid IBANs. Maybe you can work together with them to integrate that logic into Faker. That would be awesome
> 
> Mentioned Website is: http://www.randomiban.com/?country=Italy

the source code is here http://www.randomiban.com/static/mapp4.js and is pretty messy

i tend to agree with @b-pagis that i think this is out of scope for Faker. I would instead consider updating the documentation to explain that the method creates plausible but potentially invalid IBANs.
